source/websocket_decoder.c

```diff
@@ -108,27 +120,20 @@ static int s_state_extended_length(struct aws_websocket_decoder *decoder, struct
         return AWS_OP_SUCCESS;
     }
 
-    /* RFC-6455 Section 5.2 Base Framing Protocol - Payload length
-     * 1) It is a crime to use more bytes than necessary to encode length.
-     *      > in all cases, the minimal number of bytes MUST be used to encode
-     *      > the length, for example, the length of a 124-byte-long string
-     *      > can't be encoded as the sequence 126, 0, 124
-     * 2) It is a crime to use most-significant bit on 8 byte payloads.
-     *      > If 127, the following 8 bytes interpreted as a 64-bit unsigned integer
-     *      > (the most significant bit MUST be 0) are the payload length */
+    /* Actual payload length is encoded across the next 2 or 8 bytes */
     uint8_t total_bytes_extended_length;
     uint64_t min_acceptable_value;
     uint64_t max_acceptable_value;
-    if (decoder->current_frame.payload_length == VALUE_FOR_2BYTE_EXTENDED_LENGTH) {
+    if (decoder->current_frame.payload_length == AWS_WEBSOCKET_7BIT_VALUE_FOR_2BYTE_EXTENDED_LENGTH) {
         total_bytes_extended_length = 2;
-        min_acceptable_value = VALUE_FOR_2BYTE_EXTENDED_LENGTH;
-        max_acceptable_value = UINT16_MAX;
+        min_acceptable_value = AWS_WEBSOCKET_2BYTE_EXTENDED_LENGTH_MIN_VALUE;
+        max_acceptable_value = AWS_WEBSOCKET_2BYTE_EXTENDED_LENGTH_MAX_VALUE;
     } else {
-        assert(decoder->current_frame.payload_length == VALUE_FOR_8BYTE_EXTENDED_LENGTH);
+        assert(decoder->current_frame.payload_length == AWS_WEBSOCKET_7BIT_VALUE_FOR_8BYTE_EXTENDED_LENGTH);
```

Comment:
Probably shouldn't crash here, since this is directly caused by data? Instead raise an error about format and get out.

Reply:
This particular value was set by the previous state (if user passes in large value, we encode a magic number that says how many bytes we'll use to encode the real value). I'll add a comment